### PR TITLE
Add support for cmdwin

### PR DESCRIPTION
**Pull request by mityu**
I added support for cmdwin.
The basic ideas to support cmdwin are:
When the user is in cmdwin,

 - Use the alternative buffer's information (cursor position, displayed line numbers, ...).
 - Use `getcmdwintype()`/`getline()` instead of `getcmdtype()`/`getcmdline()`.
 - Specify the alternative buffer number to create extmarks/text-properties.
 - Specify the alternative buffer number to `matchaddpos()`.
 - Specify the alternative window ID to create popups on Vim.

### CODE REVIEW

File: autoload/seak.vim [hunk 1]
```diff
@@ -22,16 +30,17 @@ endfunction
...
-  let l:texts = getbufline('%', l:lnum_s, l:lnum_e)
-  let l:input = getcmdline()
-  let l:curpos = getcurpos()
-  let l:next = getcmdtype() ==# '/'
+  let l:winID = s:get_current_winID()
+  let l:lnum_s = line('w0', l:winID)
+  let l:lnum_e = line('w$', l:winID)
+  let l:texts = s:get_current_bufline(l:lnum_s, l:lnum_e)
+  let l:input = s:getcmdline()
+  let l:curpos = getcurpos(l:winID)
```
Comment (hrsh7th) [inline autoload/seak.vim]: In neovim, the getcurpos seems to not accept winID.

File: autoload/seak.vim [hunk 2]
```diff
@@ -196,7 +232,65 @@ else
...
   endfunction
 endif
 
+"
+" s:getcmdline
+"
+" Similar to the built-in getcmdline(), but cooperate with cmdwin.
+"
+function! s:getcmdline() abort
+  if getcmdline() !=# ''
```
Comment (hrsh7th) [inline autoload/seak.vim]: It should be `s:is_in_cmdwin()` instead.

File: autoload/seak.vim [hunk 3]
```diff
@@ -196,7 +232,65 @@ else
     call prop_remove({
```
Comment (hrsh7th) [inline autoload/seak.vim:232]: `call nvim_buf_del_extmark(0, ...` -> `call nvim_buf_del_extmark(s:get_current_bufnr(), ...`

**Comment by mityu** [comment]
Fixed. (Sorry for the slow response...)

File: autoload/seak.vim [hunk 4]
```diff
@@ -189,14 +225,86 @@ endif
...
+  return bufwinid(s:get_current_bufnr())
+endfunction
+
+"
+" s:getcurpos_on_current_window
+"
+if has('nvim')
+  function! s:getcurpos_on_current_window() abort
+    let l:winID = s:get_current_winID()
+    return [0, line('.', l:winID), col('.', l:winID), 0, 0]
```
Comment (hrsh7th) [inline autoload/seak.vim]: Unfortunately, the neovim doesn't accept winid on line/col too...

```
    let l:cursor = nvim_win_get_cursor(s:get_current_winID())
    return [s:get_current_bufnr(), l:cursor[0], l:cursor[1] + 1]
```

Comment (mityu) [reply autoload/seak.vim]: From `:h getpos()` and `:h getcurpos()`:
```
getpos({expr})	Get the position for String {expr}.  For possible values of
		{expr} see |line()|.  For getting the cursor position see
		|getcurpos()|.
		The result is a |List| with four numbers:
		    [bufnum, lnum, col, off]
		"bufnum" is zero, unless a mark like '0 or 'A is used, then it
		is the buffer number of the mark.
```

```
getcurpos([{winid}])
		Get the position of the cursor.  This is like getpos('.'), but
		includes an extra "curswant" item in the list:
		    [0, lnum, col, off, curswant] ~
```


According to this doc, the `bufnum` will be 0 when `'.'` is specified as `{expr}`.
So I think `s:get_current_bufnr()` is unnecessary.

Comment (hrsh7th) [reply autoload/seak.vim]:
```
    let l:cursor = nvim_win_get_cursor(s:get_current_winID())
    return [0, l:cursor[0], l:cursor[1] + 1]
```
It's OK.

Comment (mityu) [reply autoload/seak.vim]: Fixed. Thank you.

**Comment by hrsh7th** [comment]
Perfect. THank you~!

**Comment by mityu** [comment]
Thank you!

**Pull request merged**